[CI] SearchableSnapshotsBlobStoreCacheIntegTests testBlobStoreCache failed

**Build scan**:

https://gradle-enterprise.elastic.co/s/dyvp3twiy47yc

**Repro line**:

```
./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.blobstore.cache.SearchableSnapshotsBlobStoreCacheIntegTests.testBlobStoreCache" \
  -Dtests.seed=7C8C16EC4A399922 \
  -Dtests.security.manager=true \
  -Dtests.locale=uk-UA \
  -Dtests.timezone=Etc/GMT-6 \
  -Druntime.java=8
```

**Reproduces locally?**:

No

**Applicable branches**:

7.x

**Failure history**:

Just this once in the last 3 weeks:

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-21d,mode:relative,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testBlobStoreCache),sort:!(process.time-start,desc))

**Failure excerpt**:

```
жовт. 05, 2020 10:48:18 AM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
WARNING: Uncaught exception in thread: Thread[elasticsearch[node_s2][generic][T#1],5,TGRP-SearchableSnapshotsBlobStoreCacheIntegTests]
java.lang.AssertionError: SendRequestTransportException[[node_s1][127.0.0.1:42665][indices:data/read/get[s]]]; nested: NodeNotConnectedException[[node_s1][127.0.0.1:42665] connection already closed];
	at __randomizedtesting.SeedInfo.seed([7C8C16EC4A399922]:0)
	at org.elasticsearch.blobstore.cache.BlobStoreCacheService$2.onFailure(BlobStoreCacheService.java:246)
	at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50)
	at org.elasticsearch.action.support.TransportAction$1.onFailure(TransportAction.java:98)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.perform(TransportSingleShardAction.java:229)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.onFailure(TransportSingleShardAction.java:210)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.access$1100(TransportSingleShardAction.java:136)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleException(TransportSingleShardAction.java:266)
	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1181)
	at org.elasticsearch.transport.TransportService$7.doRun(TransportService.java:782)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:737)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: SendRequestTransportException[[node_s1][127.0.0.1:42665][indices:data/read/get[s]]]; nested: NodeNotConnectedException[[node_s1][127.0.0.1:42665] connection already closed];
	at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:760)
	at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:661)
	at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:593)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.perform(TransportSingleShardAction.java:246)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.start(TransportSingleShardAction.java:201)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:103)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction.doExecute(TransportSingleShardAction.java:62)
	at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:179)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:155)
	at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:83)
	at org.elasticsearch.client.node.NodeClient.executeLocally(NodeClient.java:86)
	at org.elasticsearch.client.node.NodeClient.doExecute(NodeClient.java:75)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:409)
	at org.elasticsearch.client.FilterClient.doExecute(FilterClient.java:65)
	at org.elasticsearch.client.OriginSettingClient.doExecute(OriginSettingClient.java:51)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:409)
	at org.elasticsearch.client.support.AbstractClient.get(AbstractClient.java:507)
	at org.elasticsearch.blobstore.cache.BlobStoreCacheService.getAsync(BlobStoreCacheService.java:222)
	at org.elasticsearch.blobstore.cache.BlobStoreCacheService.get(BlobStoreCacheService.java:201)
	at org.elasticsearch.index.store.SearchableSnapshotDirectory.getCachedBlob(SearchableSnapshotDirectory.java:614)
	at org.elasticsearch.index.store.cache.CachedBlobContainerIndexInput.readInternal(CachedBlobContainerIndexInput.java:198)
	at org.apache.lucene.store.BufferedIndexInput.refill(BufferedIndexInput.java:315)
	at org.apache.lucene.store.BufferedIndexInput.readByte(BufferedIndexInput.java:56)
	at org.apache.lucene.store.BufferedChecksumIndexInput.readByte(BufferedChecksumIndexInput.java:41)
	at org.apache.lucene.store.DataInput.readInt(DataInput.java:102)
	at org.apache.lucene.codecs.CodecUtil.checkHeader(CodecUtil.java:194)
	at org.apache.lucene.codecs.CodecUtil.checkIndexHeader(CodecUtil.java:255)
	at org.apache.lucene.codecs.lucene60.Lucene60FieldInfosFormat.read(Lucene60FieldInfosFormat.java:117)
	at org.apache.lucene.index.IndexWriter.readFieldInfos(IndexWriter.java:1172)
	at org.apache.lucene.index.IndexWriter.getFieldNumberMap(IndexWriter.java:1184)
	at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:1069)
	at org.elasticsearch.common.lucene.Lucene.pruneUnreferencedFiles(Lucene.java:222)
	at org.elasticsearch.repositories.blobstore.FileRestoreContext.afterRestore(FileRestoreContext.java:179)
	at org.elasticsearch.repositories.blobstore.FileRestoreContext.lambda$restore$0(FileRestoreContext.java:160)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository$9.restoreFiles(BlobStoreRepository.java:2068)
	at org.elasticsearch.repositories.blobstore.FileRestoreContext.restore(FileRestoreContext.java:156)
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$restoreShard$66(BlobStoreRepository.java:2157)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)
	... 5 more
Caused by: NodeNotConnectedException[[node_s1][127.0.0.1:42665] connection already closed]
	at org.elasticsearch.transport.TcpTransport$NodeChannels.sendRequest(TcpTransport.java:267)
	at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:747)
	... 43 more
```

--- a/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/BlobStoreCacheService.java
+++ b/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/BlobStoreCacheService.java
@@ -251,10 +251,13 @@ public class BlobStoreCacheService {
     }
 
     private static boolean isExpectedCacheGetException(Exception e) {
-        return TransportActions.isShardNotAvailableException(e)
+        if (TransportActions.isShardNotAvailableException(e)
             || e instanceof ConnectTransportException
-            || e instanceof ClusterBlockException
-            || ExceptionsHelper.unwrapCause(e) instanceof NodeClosedException;
+            || e instanceof ClusterBlockException) {
+            return true;
+        }
+        final Throwable cause = ExceptionsHelper.unwrapCause(e);
+        return cause instanceof NodeClosedException || cause instanceof ConnectTransportException;
     }
 
     public void putAsync(String repository, String name, String path, long offset, BytesReference content, ActionListener<Void> listener) {

